src/use/helpers/ImmatriculationResponseHandler.ts

```diff
@@ -1,13 +1,50 @@
 import ResponseHandler from "./ResponseHandler";
+import handleAuthenticationError from "@/api/AuthenticationHelper";
 import APIResponse from "@/api/helpers/models/APIResponse";
+import { showNetworkErrorToast, showAPIToast } from "@/use/helpers/Toasts";
 import ValidationError from "@/api/api_models/errors/ValidationError";
 import Error from "@/api/api_models/errors/Error";
 import MatriculationData from "@/api/api_models/matriculation_management/MatriculationData";
 
+export default class GenericImmatricultationResponseHandler implements ResponseHandler<boolean> {
+    handleResponse<T>(response: APIResponse<T>): T {
+        if (response.networkError) {
+            showNetworkErrorToast();
+            return response.returnValue;
+        }
+        switch (response.statusCode) {
+            case 400: {
+                showAPIToast("400");
+                return response.returnValue;
+            }
+            case 401: {
+                handleAuthenticationError(response);
+                return response.returnValue;
+            }
+            case 403: {
+                alert("You do not have the neccessary user rights for this action!");
```

Comment:
this should be changed

Reply:
this does not belong on this branch. It is just marked as added lines, because I merged two files. Please consider to open an issue for this